robolectric-resources/src/main/java/org/robolectric/res/ResType.java

```diff
@@ -49,6 +56,31 @@ private static TypedResource extractScalarItems(XpathResourceXmlLoader.XmlNode x
     return new TypedResource<>(typedResources, arrayResType, xmlContext);
   }
 
+  private static TypedResource extractTypedItems(XpathResourceXmlLoader.XmlNode xmlNode, ResType arrayResType, XmlLoader.XmlContext xmlContext) {
+    final List<TypedResource> items = new ArrayList<>();
+    for (XpathResourceXmlLoader.XmlNode item : xmlNode.selectElements("item")) {
+      final String itemString = item.getTextContent();
+      ResType itemResType = null;
```

Comment:
When I rebased I didn't notice you had added that method; it should work well here, with a few additional cases tacked on.